SA3/qa_web/views.py

```diff
@@ -1,78 +1,93 @@
-from django.contrib.auth.forms import UserCreationForm
-from django.shortcuts import render, redirect, get_object_or_404, reverse
+from django.http import HttpResponseRedirect, JsonResponse, \
+    HttpResponseForbidden
+from django.shortcuts import render, get_object_or_404
 from django.contrib.auth.decorators import login_required
 from django.views.generic import ListView
-
 from django.db.models import Count, F
 from taggit.models import Tag, TaggedItem
 from django.core.paginator import EmptyPage, PageNotAnInteger, Paginator
 from django.contrib.contenttypes.models import ContentType
-
-from django.contrib import messages
+from qa_web.forms import LoginForm, QuestionsForm, AnswersForm, EditForm, \
+    UserProfile, CustomUserCreationForm
+from qa_web.models import Answer, Question, User, Comment, Vote
 
 
-class CustomUserCreationForm(UserCreationForm):
-    class Meta(UserCreationForm.Meta):
-        model = User
-        fields = UserCreationForm.Meta.fields
-
 @login_required(login_url='/login/')
 def edit_profile(request):
-    if request.method=='GET':
+    """Displays the user profile editing form on a GET request, on a POST 
+    validates and saves the user's information.
+    Must be logged in to edit one's profile.
+
+    :param request: Request data provided by the WSGI
+    :return: Rendered template for editing user template on GET or 
+             unsuccessful validation.
+             Upon successful validation, redirects to the user's profile
+    """
+    if request.method == 'GET':
         form = UserProfile()
-        # return render_to_response('qa_web/UserProfile.html', RequestContext(request, {'form': form, }))
-        return render(request,'qa_web/EditUserProfile.html', context={'form': form})
+        return render(request, 'qa_web/edit_user_profile.html',
+                      context={'form': form})
     else:
         form = UserProfile(request.POST)
-        # validating the form occurs below and then saving the results entered by the user
         if form.is_valid():
             user = request.user
 
-            user.first_name = request.POST['prename']
-            user.last_name = request.POST['surname']
-            user.age = request.POST['age']
+            user.first_name = form.cleaned_data.get('prename')
+            user.last_name = form.cleaned_data.get('surname')
+            user.age = form.cleaned_data.get('age')
             user.email = form.cleaned_data.get('email')
             user.birthday = form.cleaned_data.get('birthday')
-            user.motherland = request.POST['motherland']
-            user.school = request.POST['school']
-            user.major = request.POST['major']
-            user.city = request.POST['city']
+            user.motherland = form.cleaned_data.get('motherland')
+            user.school = form.cleaned_data.get('school')
+            user.major = form.cleaned_data.get('major')
+            user.city = form.cleaned_data.get('city')
```

Comment:
Great to finally have `form.cleaned_data` as we have discussed many times on wanting to unify this.